src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -436,6 +442,35 @@ private void ProcessSettingsFrame(FrameHeader frameHeader)
             }
         }
 
+        private void ChangeMaxConcurrentStreams(uint newValue)
+        {
+            // The value is provided as a uint.
+            // Limit this to int.MaxValue since the CreditManager implementation only supports singed values.
+            // In practice, we should never reach this value.
+            int effectiveValue = (newValue > (uint)int.MaxValue ? int.MaxValue : (int)newValue);
+            int delta = effectiveValue - _maxConcurrentStreams;
+            _maxConcurrentStreams = effectiveValue;
+
+            _concurrentStreams.AdjustCredit(delta);
+        }
+
+        private void ChangeInitialWindowSize(int newSize)
+        {
+            Debug.Assert(newSize >= 0);
+
+            lock (_syncObject)
+            {
+                int delta = newSize - _initialWindowSize;
+                _initialWindowSize = newSize;
+
+                // Adjust existing streams
+                foreach (Http2Stream http2Stream in _httpStreams.Values)
```

Comment:
Consider instead foreach'ing over _httpStreams and then access the resulting pair's .Value.  That will be allocation free, whereas this will force the dictionary's Values collection to be instantiated.